src/main/java/com/rarchives/ripme/App.java

```diff
@@ -39,15 +39,24 @@
     private static final History HISTORY = new History();
 
     public static void main(String[] args) throws MalformedURLException {
-        Utils.configureLogger();
-        System.setProperty("apple.laf.useScreenMenuBar", "true");
-        System.setProperty("com.apple.mrj.application.apple.menu.about.name", "RipMe");
-        logger = Logger.getLogger(App.class);
-        logger.info("Initialized ripme v" + UpdateUtils.getThisJarVersion());
-
-        if (args.length > 0) {
-            handleArguments(args);
+        CommandLine cl = getArgs(args);
+        if (args.length > 0){
+          if (cl.hasOption('v')){
+            System.out.println(UpdateUtils.getThisJarVersion());
+          } else {
+            Utils.configureLogger();
+            System.setProperty("apple.laf.useScreenMenuBar", "true");
+            System.setProperty("com.apple.mrj.application.apple.menu.about.name", "RipMe");
+            logger = Logger.getLogger(App.class);
+            logger.info("Initialized ripme v" + UpdateUtils.getThisJarVersion());
+          }
+          handleArguments(args);
         } else {
+            Utils.configureLogger();
+            System.setProperty("apple.laf.useScreenMenuBar", "true");
+            System.setProperty("com.apple.mrj.application.apple.menu.about.name", "RipMe");
+            logger = Logger.getLogger(App.class);
+            logger.info("Initialized ripme v" + UpdateUtils.getThisJarVersion());
```

Comment:
Not sure quite what was ultimately achieved in the previous iteration of feedback, but you've repeated exactly the same 5 lines here in this else block as you did in the other else block above. Let's follow the DRYS (Don't Repeat YourSelf) rule here.

Let's call these five lines INIT_LOGGER

We should do initial read and parse (if applicable of the command line args). If -v then print the version, exit, and do nothing else. Otherwise, INIT_LOGGER unconditionally. Then, if args.length > 0, do command line mode, otherwise GUI.

So (rewriting, perhaps going beyond your patch to clarify what was here before):

```java
CommandLine cl = getArgs(args);
if (args.length > 0 && cl.hasOption('v')) {
    System.out.println(UpdateUtils.getThisJarVersion());
    System.exit(0); // we're done, exit early
}

// initialize logger
Utils.configureLogger();
System.setProperty("apple.laf.useScreenMenuBar", "true");
System.setProperty("com.apple.mrj.application.apple.menu.about.name", "RipMe");
logger = Logger.getLogger(App.class);
logger.info("Initialized ripme v" + UpdateUtils.getThisJarVersion());

if (args.length > 0) {
    // CLI mode
    handleArgments(args);
} else {
    // GUI mode
    MainWindow mw = new MainWindow();
    SwingUtilities.invokeLater(mw);
}
```

Reply:
@metaprime Done as you proposed. Thanks for the feedback